webapi/src/main/scala/org/knora/webapi/responders/v2/SearchResponderV2.scala

```diff
@@ -555,16 +555,60 @@ class SearchResponderV2 extends ResponderWithStandoffV2 {
           * @param expression           the transformed Filter expression. In some cases, a given FILTER expression is replaced by additional statements.
           * @param additionalStatements additionally created statement patterns.
           */
-        protected case class TransformedFilterExpression(expression: Option[Expression], additionalStatements: Seq[StatementPattern] = Seq.empty[StatementPattern])
+        protected case class TransformedFilterPattern(expression: Option[Expression], additionalStatements: Seq[StatementPattern] = Seq.empty[StatementPattern])
+
+        /**
+          * Handles query variables that represent properties in a [[FilterPattern]].
+          *
+          * @param queryVar the query variable to be handled.
+          * @param comparisonOperator the comparison operator used in the filter expression.
+          * @param iriRef the Iri the property query variable is restricted to.
+          * @param propInfo information about the query variable's type.
+          * @return a [[TransformedFilterPattern]].
+          */
+        private def handlePropertyIriQueryVar(queryVar: QueryVariable, comparisonOperator: CompareExpressionOperator.Value, iriRef: IriRef, propInfo: PropertyTypeInfo): TransformedFilterPattern = {
```

Comment:
@benjamingeer Do you think it is ok to use `private` for these methods? The method will be only used within the same "template" (https://www.scala-lang.org/files/archive/spec/2.11/05-classes-and-objects.html#private).

also see https://stackoverflow.com/questions/44747514/how-to-inherit-a-protected-private-of-an-abstract-class-in-scala


https://stackoverflow.com/questions/44747514/how-to-inherit-a-protected-private-of-an-abstract-class-in-scala

Reply:
Yes, if subclasses don't need to call those methods, it's better for them to be private.